$OpenBSD: patch-src_plugin_setup_cpp,v 1.4 2011/01/18 11:04:23 dcoppa Exp $

fix from upstream: set width and height from the style attribute
if present

--- src/plugin_setup.cpp.orig	Fri Nov  5 17:12:59 2010
+++ src/plugin_setup.cpp	Tue Jan 18 11:07:47 2011
@@ -112,6 +112,14 @@ void new_instance(CPlugin * instance, int16_t argc, ch
             if (g_ascii_strcasecmp(argn[i], "height") == 0) {
                 sscanf(argv[i], "%i", &height);
             }
+            
+            if (g_ascii_strcasecmp(argn[i], "style") == 0) {
+                tmp = g_strrstr(argv[i],"width:");
+                sscanf(tmp + strlen("width:"), "%i", &width);
+                tmp = g_strrstr(argv[i],"height:");
+                sscanf(tmp + strlen("height:"), "%i", &height);
+            }
+            
 
             if (g_ascii_strcasecmp(argn[i], "src") == 0 || g_ascii_strcasecmp(argn[i], "url") == 0) {
                 item = g_new0(ListItem, 1);
